src/zc/buildout/easy_install.py

```diff
@@ -733,7 +736,29 @@ def install(self, specs, working_set=None):
                     pkg_resources.VersionConflict(dist, req), ws)
 
             best[req.key] = dist
-            extra_requirements = dist.requires(req.extras)[::-1]
+
+            missing_requested = sorted(
+                set(req.extras) - set(dist.extras)
+            )
+            for missing in missing_requested:
+                logger.warning(
+                    '%s does not provide the extra \'%s\'',
+                    dist, missing
+                )
+
+            if missing_requested:
+                if not self._allow_unknown_extras:
+                    raise zc.buildout.UserError(
+                        "Couldn't find the required extras. Add 'allow-unknown-extras=true' "
+                        "to the [buildout] configuration if this is acceptable."
```

Comment:
I'll prefer some different wording.

"Couldn't find the required extra.  This means the requirement is incorrect.  If the requirement is itself from software you requested, then there might be a bug in requested software.  You can ignore this by using  'allow-unknown-extras=true', however that may simply cause needed software to be omitted."

Reply:
After all, we're not doing this to be mean and make people say the magic word.